File: snuggle_server/src/main/java/com/ssafy/snuggle/model/service/AddressService.java
package com.ssafy.snuggle.model.service;

import com.ssafy.snuggle.model.dto.Address;

public interface AddressService {

	/**
	 * 사용자 id로 주소 조회한다.
	 * 
	 * @param userId
	 */
	Address select(String userId);

	/**
	 * 사용자의 주소 수정 userId로 접근
	 * 
	 * @param userId
	 */
	public int update(Address address);
}

File: snuggle_final_app/app/src/main/java/com/ssafy/snuggle_final_app/data/service/AddressService.kt
package com.ssafy.snuggle_final_app.data.service

import com.ssafy.snuggle_final_app.data.model.dto.Address
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface AddressService {

    @PUT("/snuggle/address")
    suspend fun updateAddress(@Body address: Address): Boolean

    @POST("/snuggle/address")
    suspend fun insertAddress(@Body address: Address): Int

    @GET("/snuggle/address/{userId}")
    suspend fun getAddress(@Path("userId") userId: String): Address

    @GET("/snuggle/address/isExist/{userId}")
    suspend fun isExist(@Path("userId") userId: String): Boolean
}